### PR TITLE
Add support for zwave-js 12.0 and schema 32

### PR DESCRIPTION
**Pull request by raman325**
## Breaking Changes

Per the [Z-Wave JS 12.x migration guide](https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v12), many commands in the driver have been renamed, and the corresponding model methods/properties and events have changed accordingly:
- `Controller.is_heal_network_active` property → `Controller.is_rebuilding_routes`
- `Controller.async_begin_healing_network` method → `Controller.async_begin_rebuilding_routes`
- `Controller.async_stop_healing_network` method → `Controller.async_stop_rebuilding_routes`
- `Controller.async_heal_node` method → `Controller.async_rebuild_node_routes`
- `heal network progress` controller event -> `rebuild routes progress`
- `heal network done` controller event -> `rebuild routes done`

Additionally, `Controller.async_firmware_update_ota` now requires `list[NodeFirmwareUpdateInfo]` instead of `list[NodeFirmwareUpdateFileInfo]` as an update

## Description

Follows https://github.com/zwave-js/zwave-js-server/pull/1036

Migration guide: https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v12

### CODE REVIEW

**Review by MartinHjelmare — COMMENTED** [review]
Can we add tests for the non covered lines?

**Comment by raman325** [comment]
> Can we add tests for the non covered lines?

Once the PR linked above gets merged, I will refactor this one and add the missing coverage, we should be able to get this PR to 100%

**Review by MartinHjelmare — APPROVED** [review]

**Comment by MartinHjelmare** [comment]
Please mention the breaking changes in the client in the PR description.

File: zwave_js_server/exceptions.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def __init__(
         self, message_id: str, error_code: str, msg: str | None = None
     ) -> None:
         """Initialize a failed command error."""
-        super().__init__(msg or f"Command failed: {error_code}")
+        super().__init__(f"{error_code}: {msg}" or f"Command failed: {error_code}")
```
Comment (MartinHjelmare) [inline zwave_js_server/exceptions.py:89]: We can't write the condition like that now since the first part is always true.

**Comment by raman325** [comment]
> Please mention the breaking changes in the client in the PR description.

You mean the fact that the `FailedCommand` exception parameters have changed?

File: zwave_js_server/exceptions.py (hunk 1 above)
Comment (raman325) [reply zwave_js_server/exceptions.py:89]: good point

**Comment by MartinHjelmare** [comment]
No those haven't changed. We've removed properties and methods and added other ones to replace them regarding healing and changed a parameter in `async_firmware_update_ota` in the controller model.

**Comment by raman325** [comment]
> No those haven't changed. We've removed properties and methods and added other ones to replace them regarding healing and changed a parameter in `async_firmware_update_ota` in the controller model.

ah thanks, thought we were talking about the client module 😅

**Pull request merged**